.github/workflows/verify_samples.yml

```diff
@@ -0,0 +1,51 @@
+name: Build Samples
+
+on: 
+  push:
+    branches: master
+  pull_request:
+    branches: '*'
+
+jobs:
+  linux:
+    name: Ubuntu
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          submodules: recursive
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update -y && sudo apt-get install -y flex bison clang lld
+      - name: Build
+        run: sh ./.ci_build_samples.sh
+  macOS:
+    name: macOS
+    runs-on: macOS-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          submodules: recursive
+      - name: Install Dependencies
+        run: |
+          brew update && brew install llvm
+      - name: Build
+        run: sh ./.ci_build_samples.sh
+  windows:
+    name: Windows
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          submodules: recursive
+      - name: Install MSYS2 & Dependencies
+        run: |
+          choco install msys2
+          C:\tools\msys64\usr\bin\bash.exe -lc "pacman --needed --noconfirm -S make bison flex mingw-w64-x86_64-llvm"
+          C:\tools\msys64\usr\bin\bash.exe -lc "pacman --needed --noconfirm -S mingw-w64-x86_64-clang mingw-w64-x86_64-lld"
```

Comment:
Why two separate invocations? Also this will have to install CMake now, too.

Reply:
I have it be 2 different calls to avoid a needlessly long line (although it's already over 80col so not sure why I care), and yes I just noticed we updated the exiso submodule.

If you want, again, I can change this.

Reply:
Change it.

Two invocations might affect behavior (dependencies for example) and should be avoided on CI, unless there's actual reasons as to why it's necessary.

There's probably also a way to split lines in the string (typically a backslash) if it's really necessary.

Reply:
Fixed.